exercises/tutorial_halfday/ex7_tiled-matrix-transpose.cpp

```diff
@@ -0,0 +1,288 @@
+{
+
+  std::cout << "\n\nExercise #7: RAJA Tiled Matrix Transpose...\n";
+
+  //
+  // Define num rows/cols in matrix
+  //
+  const int N_r = 56;
+  const int N_c = 75;
+
+  //
+  // Allocate matrix data
+  //
+  int *A = memoryManager::allocate<int>(N_r * N_c);
+  int *At = memoryManager::allocate<int>(N_r * N_c);
+
+  //
+  // In the following implementations of tiled matrix transpose, we
+  // use RAJA 'View' objects to access the matrix data. A RAJA view
+  // holds a pointer to a data array and enables multi-dimensional indexing
+  // into the data.
+  //
+  RAJA::View<int, RAJA::Layout<DIM>> Aview(A, N_r, N_c);
+  RAJA::View<int, RAJA::Layout<DIM>> Atview(At, N_c, N_r);
+
+  //
+  // Define TILE dimensions
+  //
+  const int TILE_DIM = 16;
+
+  // Calculate number of tiles (needed for the c version)
+  const int outer_Dimc = (N_c - 1) / TILE_DIM + 1;
+  const int outer_Dimr = (N_r - 1) / TILE_DIM + 1;
+
+  //
+  // Initialize matrix data
+  //
+  for (int row = 0; row < N_r; ++row) {
+    for (int col = 0; col < N_c; ++col) {
+      Aview(row, col) = col;
+    }
+  }
+  //printResult<int>(Aview, N_r, N_c);
+
+  //----------------------------------------------------------------------------//
+  std::cout << "\n Running C-version of tiled matrix transpose...\n";
+
+  std::memset(At, 0, N_r * N_c * sizeof(int));
+
+  //
+  // (0) Outer loops to iterate over tiles
+  //
+  for (int by = 0; by < outer_Dimr; ++by) {
+    for (int bx = 0; bx < outer_Dimc; ++bx) {
+      //
+      // (1) Loops to iterate over tile entries
+      //
+      for (int ty = 0; ty < TILE_DIM; ++ty) {
+        for (int tx = 0; tx < TILE_DIM; ++tx) {
+
+          int col = bx * TILE_DIM + tx;  // Matrix column index
+          int row = by * TILE_DIM + ty;  // Matrix row index
+
+          // Bounds check
+          if (row < N_r && col < N_c) {
+            Atview(col, row) = Aview(row, col);
+          }
+        }
+      }
+
+    }
+  }
+
+  checkResult<int>(Atview, N_c, N_r);
+  // printResult<int>(Atview, N_c, N_r);
+  //----------------------------------------------------------------------------//
+
+  //
+  // The following RAJA variants will use the RAJA::kernel method to carryout the
+  // transpose.
+  //
+  // Here, we define RAJA range segments to establish the iteration spaces.
+  // Further partioning of the iteration space is carried out in the
+  // tile_fixed statements. Iterations inside a RAJA loop is given by their
+  // global iteration number.
+  //
+  RAJA::RangeSegment row_Range(0, N_r);
+  RAJA::RangeSegment col_Range(0, N_c);
+
+  //----------------------------------------------------------------------------//
+  std::cout << "\n Running sequential tiled matrix transpose ...\n";
+  std::memset(At, 0, N_r * N_c * sizeof(int));
+
+  //
+  // The following policy carries out the transpose
+  // using sequential loops. The template parameter inside
+  // tile_fixed corresponds to the dimension size of the tile.
+  //
+  using KERNEL_EXEC_POL_SEQ =
```

Comment:
The policy should be Ifdef'd out so the CI tests compile. Then, note below that they have to turn it on.